src/plugins/csp_configuration_provider/server/provider.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { OpenSearchClient } from '../../../../src/core/server';
+
+import { CspClient } from './types';
+
+export class OpenSearchCspClient implements CspClient {
+  private client: OpenSearchClient;
+
+  constructor(inputOpenSearchClient: OpenSearchClient) {
+    this.client = inputOpenSearchClient;
+  }
+
+  async exists(configurationName: string): Promise<boolean> {
+    let value = false;
+
+    try {
+      const exists = await this.client.indices.exists({
+        index: configurationName,
+      });
+
+      value = exists.body;
+    } catch (e) {
+      // eslint-disable-next-line no-console
+      console.error(
```

Comment:
Why output to console ?

Reply:
will pass in `logger` and remove console.